### ISSUE TITLE
working directory of Rmd file different in the visual editor

### ISSUE THREAD

**Issue by cderv** [rstudio/rstudio#7582]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.708 
    OS Version      :  WINDOWS 10
    R Version       :  4.0.2

### Steps to reproduce the problem

1. Create new Rstudio Project
2. Create a new folder : `dir.create("rmd")`
3. Create a new Rmd file : `file.create("rmd/test.Rmd")`
4. Copy this into the document: `file.edit("rmd/test.Rmd")` then Copy paste
````md
---
title: test
output: html_document
---

```{r}
getwd()
```
````
and save
5. **In a non visual editor mode**, Run the chunk: The working directory is document directory as expected
6. Switch to visual editor mode, Run the chunk: The working directory is not the same, and seems to be the project directory.
7. In the R console, set another working dir, like `setwd(tempdir())`
8. change option "Knit > Knit Directory > Current Working Dir 
9. In both mode, the option is taken into account. 
10. Go back to "Knit > Knit Directory > Document Directory"
11. In non visual editing the working dir for the chunks is correct, in visual editing mode it is still the temp directory. 

### Describe the problem in detail

The working directory not being the same, the code chunk behave differently regarding files when running from the visual editor or not. This seems unexpected. All the more because the RStudio setting is not respected: 
![image](https://user-images.githubusercontent.com/6791940/90235615-e8d85480-de21-11ea-9f00-7ab0aa316c30.png)

The document directory option seems not taken into account by the visual editor. (see 7 to 11 above)

When knitting with the button from the visual editor it works as expected. 

### Describe the behavior you expected

I would expect by default that the working directory for an Rmd file is the Document directory, and most of all to be the same when I execute chunk from the visual editor or not. 
I also expect the options to be taken into account by the visual editor mode.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by jjallaire** [comment]
@jmcphers I believe that this will be addressed by https://github.com/rstudio/rstudio/pull/7666, correct?

**Comment by jmcphers** [comment]
That's correct, executing chunks in visual mode should be symmetric with code view once that PR is merged.

**Comment by jjallaire** [comment]
Addressed via https://github.com/rstudio/rstudio/pull/7666

**Comment by R-Hannibal** [comment]
@cderv, thanks again for finding and reporting this issue. 

We have a fix in place in our latest Daily build of the upcoming 1.4 release of RStudio (available [HERE](https://dailies.rstudio.com/)). If you get the chance will you let us know if the issue is resolved for you in that build?

**Comment by cderv** [comment]
I have just tested with 1.4.792 and this now works as expected regarding knitr working dir. 
I have just found another issue in the process of replaying this test. I'll report it. 

Thanks !
